contracts/LockScheme.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.12;
+
+import "openzeppelin-solidity/contracts/math/Math.sol";
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "./interfaces/IERC20Detailed.sol";
+import "./SafeERC20Detailed.sol";
+import "./PercentageCalculator.sol";
+
+contract LockScheme {
+
+    using SafeMath for uint256;
+    using SafeERC20Detailed for IERC20Detailed;
+
+    uint256 public lockBlock; // The period of lock for this contract
```

Comment:
The lock period is for each user and depends on when the user gets into the staking.